File: app/src/main/java/com/example/imhere/LoginActivity.java
package com.example.imhere;

import androidx.appcompat.app.AppCompatActivity;

import android.content.ContentValues;
import android.content.SharedPreferences;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class LoginActivity extends AppCompatActivity implements View.OnClickListener {

    Button loginButton;

    EditText loginText;
    private EditText passwordText;

    DataBaseHelper dbHelper;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        loginButton = findViewById(R.id.loginButton);

        loginText = findViewById(R.id.loginText);
        passwordText = findViewById(R.id.passwordText);

        loginButton.setOnClickListener(this);

        dbHelper = new DataBaseHelper(this);
    }

    @Override
    public void onClick(View v) {
        String loginStr = loginText.getText().toString();
        int hashCode = passwordText.getText().toString().hashCode();
        SQLiteDatabase db = dbHelper.getWritableDatabase();
        Cursor c = db.query("accountTable", null, "login == ?", new String[] { loginStr }, null, null, null);
        if (c == null || c.getCount() == 0) {
            Toast.makeText(this, "Неверный логин или пароль", Toast.LENGTH_LONG).show();
            return;
        }
        c.moveToFirst();
        int inputHashCode = c.getInt(c.getColumnIndex("password"));
        if (inputHashCode != hashCode){
            Toast.makeText(this, "Неверный логин или пароль", Toast.LENGTH_LONG).show();
            return;
        }
        c.close();
        dbHelper.close();
        SharedPreferences sp = getSharedPreferences("autoriz", MODE_PRIVATE);
        SharedPreferences.Editor ed = sp.edit();
        ed.putBoolean("auton", true);
        ed.putBoolean("task", true);
        ed.apply();
        super.finish();
    }

    @Override
    public void onBackPressed(){
    }
}

File: app/src/main/java/com/example/imhere/LoginActivity.kt
package com.example.imhere

import android.content.Context
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class LoginActivity : AppCompatActivity(), View.OnClickListener {
    lateinit var loginButton: Button
    private var loginText: EditText? = null
    private var passwordText: EditText? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        loginButton = findViewById(R.id.loginButton)
        loginText = findViewById(R.id.loginText)
        passwordText = findViewById(R.id.passwordText)
        loginButton.setOnClickListener(this)
    }

    /*По клику - проверка данных с данными в базе данных
    * Механизм на случай, если не выйдет с ЛК
    * + готовая форма аутентификации*/
    override fun onClick(v: View) {
        val loginStr = loginText!!.text.toString()
        val passwordHashCode = passwordText!!.text.toString().hashCode()
        val dbHelper = DataBaseHelper(this)
        val db = dbHelper.writableDatabase
        val c = db.query("accountTable", null, "login == ?", arrayOf(loginStr), null, null, null)
        if (c == null || c.count == 0) {
            Toast.makeText(this, "Неверный логин или пароль", Toast.LENGTH_LONG).show()
            return
        }
        c.moveToFirst()
        val rightPasswordHashCode = c.getInt(c.getColumnIndex("password"))
        if (passwordHashCode != rightPasswordHashCode) {
            Toast.makeText(this, "Неверный логин или пароль", Toast.LENGTH_LONG).show()
            return
        }
        c.close()
        dbHelper.close()
        val sp = getSharedPreferences("authentication", Context.MODE_PRIVATE)
        val ed = sp.edit()
        ed.putBoolean("authentication", true)
        ed.apply()
        super.finish()
    }

    override fun onBackPressed() {
    }
}